tests/test_translations.h

```diff
@@ -0,0 +1,75 @@
+/*************************************************************************/
+/* Copyright (c) 2007-2021 Juan Linietsky, Ariel Manzur.                 */
+/* Copyright (c) 2014-2021 Godot Engine contributors (cf. AUTHORS.md).   */
+/*                                                                       */
+/* Permission is hereby granted, free of charge, to any person obtaining */
+/* a copy of this software and associated documentation files (the       */
+/* "Software"), to deal in the Software without restriction, including   */
+/* without limitation the rights to use, copy, modify, merge, publish,   */
+/* distribute, sublicense, and/or sell copies of the Software, and to    */
+/* permit persons to whom the Software is furnished to do so, subject to */
+/* the following conditions:                                             */
+/*                                                                       */
+/* The above copyright notice and this permission notice shall be        */
+/* included in all copies or substantial portions of the Software.       */
+/*                                                                       */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,       */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF    */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.*/
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY  */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,  */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE     */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
+/*************************************************************************/
+
+#ifndef TEST_TRANSLATIONS_H
+#define TEST_TRANSLATIONS_H
+
+#include "core/io/resource_loader.h"
+#include "core/os/memory.h"
+#include "core/string/translation.h"
+#include "editor/import/resource_importer_csv_translation.h"
+
+namespace TestTranslations {
+
+TEST_CASE("[Translation] CSV import") {
+	TranslationServer *translation_server = nullptr;
+	translation_server = memnew(TranslationServer);
+	translation_server->setup(); //register translations, load them, etc.
+	translation_server->load_translations();
+
+	Ref<ResourceImporterCSVTranslation> import_csv_translation;
+	import_csv_translation.instance();
+
+	Map<StringName, Variant> options;
+	options["compress"] = false;
+	options["delimiter"] = 0;
+
+	List<String> gen_files;
+
+	Error result = import_csv_translation->import(TestUtils::get_data_path("translations.csv"),
+			"", options, nullptr, &gen_files);
+	CHECK(result == OK);
+	CHECK(gen_files.size() == 2);
+
+	for (List<String>::Element *E = gen_files.front(); E; E = E->next()) {
+		Ref<Translation> translation = ResourceLoader::load(E->get());
+		CHECK(translation.is_valid());
+		TranslationServer::get_singleton()->add_translation(translation);
```

Comment:
That's what I'm talking about, you either have to use `TranslationServer::get_singleton()` or instantiate another server (which goes unused in this test case it seems), but I'd just use `TranslationServer::get_singleton()`.